tableaudocumentapi/datasource.py

```diff
@@ -141,6 +178,16 @@ def fields(self):
         return self._fields
 
     def _get_all_fields(self):
-        column_objects = (_mapping_from_xml(self._datasourceTree, xml)
-                          for xml in self._datasourceTree.findall('.//column'))
-        return MultiLookupDict({k: v for k, v in column_objects})
+        column_objects = [_column_object_from_column_xml(self._datasourceTree, xml)
+                          for xml in self._datasourceTree.findall('.//column')]
+        existing_fields = [x.id for x in column_objects]
+        metadata_fields = (x.text
+                           for x in self._datasourceTree.findall(".//metadata-record[@class='column']/local-name"))
+
+        missing_fields = (x for x in metadata_fields if x not in existing_fields)
```

Comment:
Looking very clean!

It might help readability if we break this into some more functions to abstract out the metadata processing a bit.

In psuedocode:
```
_get_all_fields():
    all_the_fields = []
    all_the_fields += _get_fields_from_column_xml()
    all_the_fields += _get_weird_metadata_only_fields()

    return FieldDictionary(k: v for k, v in all_the_fields)

def _get_weird_metadata_only_fields():
    does the list comprehension
```
Also fine to split that into a different issue/PR